### PR TITLE
Plugin TR-064

### PR DESCRIPTION
**Pull request by ChristianMayer**
Plugin for TR-064 connection to a router like the Fritz!Box to show e.g. the call list.

### CODE REVIEW

File: source/class/cv/plugins/TR064/calllist.js [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+&location=upnp/control/x_tam&uri=urn:dslforum-org:service:X_AVM-DE_TAM:1&fn=GetMessageList&p[]=NewIndex&v[]=0
+&location=upnp/control/x_contact&uri=urn:dslforum-org:service:X_AVM-DE_OnTel:1&fn=GetInfoByIndex&p[]=NewIndex&v[]=1
+&location=upnp/control/x_contact&uri=urn:dslforum-org:service:X_AVM-DE_OnTel:1&fn=GetCallList
+&location=upnp/control/wlanconfig1&uri=urn:dslforum-org:service:WLANConfiguration:1&fn=GetInfo
+
+&location=&uri=&fn=&p[]=&v[]=
+
+*/
+qx.Class.define('cv.plugins.TR064.calllist', {
+  extend: cv.ui.structure.AbstractWidget,
```
Comment (peuter) [inline source/class/cv/plugins/TR064/calllist.js]: Just one thing about "package" and Class naming conventions:
- The currently used convention is: lower-case package names and CamelCase class names. This has been used throughout the whole CometVisu and we should stick to it!

So please rename this one to `cv.plugins.tr064.CallList`

File: source/resource/visu_config.xsd [hunk 2]
```diff
@@ -1862,6 +1863,117 @@
...
+      <xsd:annotation>
+        <xsd:documentation xml:lang="en">Icon for playing the telephone answering machine.</xsd:documentation>
+        <xsd:documentation xml:lang="de">Icon für Anrufbeantworter abspielen.</xsd:documentation>
+      </xsd:annotation>
+    </xsd:attribute>
+    <xsd:attribute name="TAMplayColor" type="xsd:string" use="optional" />
+    <xsd:attribute name="TAMstop" type="xsd:string" use="optional">
+      <xsd:annotation>
+        <xsd:documentation xml:lang="en">Icon for stopping the telephone answering machine.</xsd:documentation>
+        <xsd:documentation xml:lang="de">Icon für Anrufbeantworter Stop.</xsd:documentation>
```
Comment (peuter) [inline source/resource/visu_config.xsd]: Typo => `stopp`

File: source/class/cv/plugins/TR064/calllist.js [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+  ******************************************************
+  */
+  members: {
+    __calllistUri: '',
+    __calllistList: undefined,
+    __refreshingCalllist: false,
+    __TAMeventAttached: {},
+    
+    _getInnerDomString: function () {
+      this.update();
```
[the remark this answers is not in the document]
Comment (ChristianMayer) [reply source/class/cv/plugins/TR064/calllist.js]: It's here to get the content from the router and reuse the `update()` code for that.
But as that's misleading I'm changing it to `refreshCalllist()` - although it's not a "refresh" it's more clear about the intentions

File: source/class/cv/plugins/TR064/calllist.js [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+        if( !this.__refreshingCalllist )
+          this.refreshCalllist('timer');
+      }, this);
+      this._timer.start();
+    },
+    _update: function(address, value) {
+      if( !this.__refreshingCalllist )
+      {
+        this.refreshCalllist('update');
+        return;
```
[the remark this answers is not in the document]
Comment (ChristianMayer) [reply source/class/cv/plugins/TR064/calllist.js]: correct

File: source/class/cv/plugins/TR064/calllist.js [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+          }
+        });
+        html += '</tr>';
+      });
+      clLi.innerHTML = html;
+      //// ES6:
+      // for( let tam of clLi.getElementsByClassName('tam') )
+      //   tam.addEventListener("click", function(){ self.__playTAM(this); } );
+      //// ES5:
+      var tamList = clLi.getElementsByClassName('tam');
```
[the remark this answers is not in the document]
Comment (ChristianMayer) [reply source/class/cv/plugins/TR064/calllist.js]: It's a hint for the improvement that is possible as soon as we are not blocked away from ES6 by Qooxdoo anymore...

File: source/class/cv/plugins/TR064/calllist.js [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+          for( let item of data.getElementsByTagName('Call') ) {
+            var entry = {};
+            for( let node of item.children ) {
+              entry[node.nodeName] = node.textContent;
+            }
+            self.__calllistList.push( entry );
+          }
+          ---
+          ES5:
+          */
```
[the remark this answers is not in the document]
Comment (ChristianMayer) [reply source/class/cv/plugins/TR064/calllist.js]: It's a hint for the improvement that is possible as soon as we are not blocked away from ES6 by Qooxdoo anymore...

File: source/class/cv/plugins/TR064/calllist.js (hunk 1 above)
Comment (ChristianMayer) [reply source/class/cv/plugins/TR064/calllist.js]: Everything is renamed now

**Comment by ChristianMayer** [comment]
All points from the review should be addressed now

File: source/class/cv/plugins/TR064/calllist.js (hunk 1 above)
Comment (peuter) [reply source/class/cv/plugins/TR064/calllist.js]: The classname itself is still lowercase and no camelcase: should be `cv.plugins.tr064.CallList`

Comment (peuter) [reply source/class/cv/plugins/TR064/calllist.js]: and don't forget to change the filename too then (CallList.js)

Comment (ChristianMayer) [reply source/class/cv/plugins/TR064/calllist.js]: Done

**Comment by peuter** [comment]
Nitpicking mode:

there are several occurences of `incomming` (Schema, SourceCode, ...) they should be replaced with `incoming`

**Comment by peuter** [comment]
Ok I'm done with my review now!

**Comment by ChristianMayer** [comment]
@peuter "the cursor should be changed to pointer (afair) to show the user that this icon is clickable."
=> Currently all actors are NOT changing the cursor type (checked with pure) - except traditional links
This is most likely a left over

But: I think we should have a general discussion and agreement whether we want `pointer` on any "modifyable"/interactable thing (like the links currently) or not change the cursor at all.

=> What's you position on that topic?

**Comment by ChristianMayer** [comment]
@peuter Please check you last check mark comment as it might be solved now and I couldn't reproduce it.

Except my pointer question (see above) that concerns should be addressed now.

**Comment by peuter** [comment]
Regarding Pointer:
I think its a design question: The metal-Design does use pointer-cursors whereever something is clickable. I wasn't aware yet that pure doesn' t do that.

My personal opinion is that the user should be given some kind of hint whats clickable or not. Doing this by changing the cursor is a common way to achieve that. Might not be needed if the widget looks like a button, but in other cases you need something like that.

If you include the changes to the test i posted on gitter I will merge this PR.

**Pull request merged**